temperusb/cli.py

```diff
@@ -1,102 +1,56 @@
-    sensor_id = [0,]  # Default to first sensor unless specified
-    disp_ports = False
-    quiet_output = False
-    for o, a in opts:
-        if o == "-p":
-            disp_ports = True
-        elif o == "-c":
-            degree_unit = 'c'
-        elif o == "-f":
-            degree_unit = 'f'
-        elif o == "-q":
-            quiet_output = True
-        elif o == "-s":
-            if a == "all":
-                sensor_id = "all"
-            else:
-                try:
-                    sensor_id = int(a)
-                    if not (sensor_id == 0 or sensor_id == 1):
-                        raise ValueError(
-                            "sensor_id should be 0 or 1, %d given" % sensor_id
-                            )
-                    # convert to list
-                    sensor_id = [sensor_id,]
-                except ValueError as err:
-                    print(str(err))
-                    usage()
-                    sys.exit(3)
-        elif o == "-S":
-            try:
-                sensor_count = int(a)
-                if not (sensor_count == 1 or sensor_count == 2):
-                    raise ValueError(
-                        "sensor_count should be 1 or 2, %d given" % sensor_count
-                        )
-            except ValueError as err:
-                print(str(err))
-                usage()
-                sys.exit(4)
-        elif o in ("-h", "--help"):
-            usage()
-            sys.exit()
-        else:
-            raise RuntimeError("Unhandled option '%s'" % o)
 
-    if quiet_output and not degree_unit:
-        print('You need to specify unit (-c of -f) when using -q option')
-        sys.exit(1)
+def parse_args():
+    descr = "Temperature data from a TEMPer v1.2 sensor."
+
+    parser = argparse.ArgumentParser(description=descr)
+    parser.add_argument("-p", "--disp_ports", action='store_true',
+                        help="Display ports")
+    units = parser.add_mutually_exclusive_group(required=False)
+    units.add_argument("-c", "--celsius", action='store_true',
+                       help="Quiet: just degrees celcius as decimal")
+    units.add_argument("-f", "--fahrenheit", action='store_true',
+                       help="Quiet: just degrees fahrenheit as decimal")
+    parser.add_argument("-s", "--sensor_ids", choices=['0', '1', 'all'],
+                        help="IDs of sensors to use on the device " +
+                        "(multisensor devices only)", default='0')
+    parser.add_argument("-S", "--sensor_count", choices=[1, 2], type=int,
+                        help="Specify the number of sensors on the device",
+                        default='1')
+    args = parser.parse_args()
 
-    # handle the sensor_id "all" option - convert to number of sensors
-    if sensor_id == "all":
-        sensor_id = range(0, sensor_count)
+    args.sensor_ids = list(range(args.sensor_count)) if args.sensor_ids == 'all' \
```

Comment:
I'd prefer a "traditional" if-statement here for consistency...

```
if args.sensor_ids == 'all':
    args.sensor_ids = range(args.sensor_count)
else:
    args.sensor_ids = [int(args.sensor_ids)]
```

What do others think?


(also, ```range()``` returns a list, no need for ```list()```)